### PR TITLE
KOGITO-1293: Handled 404 error and URL parameters

### PR DESCRIPTION
**Pull request by Sara4994**

### CODE REVIEW

**Review by cristianonicolai — APPROVED** [review]

**Comment by srambach** [comment]
My only comment is that "Navigate to home" could be clearer. @kywalker-rh and I were thinking "Go to process instances" or "Go to domain explorer" if you can differentiate. "Go" is simpler than "Navigate" and there isn't a clear "home" here.
Otherwise, 👍

**Comment by cristianonicolai** [comment]
@Sara4994 I guess the idea with home, is that we go back to the landing page of the app and that could change over time. With the 404 error page I dont think we can determine what the user was look for. Then for the process instance not found page, I agree, we could change to  something like "Go/Navigate to Process instance list page"

**Comment by Sara4994** [comment]
> @Sara4994 I guess the idea with home, is that we go back to the landing page of the app and that could change over time. With the 404 error page I dont think we can determine what the user was look for. Then for the process instance not found page, I agree, we could change to something like "Go/Navigate to Process instance list page"

@cristianonicolai @srambach done

File: packages/management-console/src/components/Molecules/ErrorComponent/ErrorComponent.tsx [hunk 1]
```diff
@@ -1,52 +1,56 @@
...
+            <EmptyStateIcon
+              icon={ExclamationCircleIcon}
+              size="md"
+              color="var(--pf-global--danger-color--100)" />
+            <Title headingLevel="h1" size="4xl">404 Error: page not found</Title>
+            <EmptyStateBody>
+              This page could not be found.
+            </EmptyStateBody>
+            <Button variant="primary" onClick={redirectHandler}>
+              Go to process list
```
Comment (cristianonicolai) [inline packages/management-console/src/components/Molecules/ErrorComponent/ErrorComponent.tsx]:
```suggestion
              Go to Process Instances
```

File: packages/management-console/src/components/Templates/DomainExplorerDashboard/DomainExplorerDashboard.tsx [hunk 2]
```diff
@@ -184,7 +184,13 @@ const DomainExplorerDashboard = props => {
 
   return (
     <>
-      {!props.domains.includes(domainName) && <Redirect to="/ErrorComponent" />}
+      {!props.domains.includes(domainName) && <Redirect to={{
+        pathname: '/NoData', state: {
+          prev: location.pathname,
+          title: 'Domain not found', description: `Domain with the name ${domainName} not found`,
+          buttonText: 'Go to domain list'
```
Comment (cristianonicolai) [inline packages/management-console/src/components/Templates/DomainExplorerDashboard/DomainExplorerDashboard.tsx]:
```suggestion
          buttonText: 'Go to Domain Explorer'
```

File: packages/management-console/src/components/Templates/ProcessDetailsPage/ProcessDetailsPage.tsx [hunk 3]
```diff
@@ -23,6 +23,19 @@ const ProcessDetailsPage = ({ match }) => {
...
   });
 
+  if (data) {
+    const result = data.ProcessInstances;
+    if (result.length === 0) {
+      return <Redirect to={{
+        pathname: '/NoData', state: {
+          prev: location.pathname,
+          title: 'Process not found', description: `Process instance with the id ${id} not found`,
+          buttonText: 'Go to process list'
```
Comment (cristianonicolai) [inline packages/management-console/src/components/Templates/ProcessDetailsPage/ProcessDetailsPage.tsx]:
```suggestion
          buttonText: 'Go to Process Instances'
```

File: packages/management-console/src/components/Molecules/NoDataComponent/NoDataComponent.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+            <PageSection isFilled={true}>
+                <Bullseye>
+                    <EmptyState variant={EmptyStateVariant.full}>
+                        <EmptyStateIcon icon={SearchIcon} />
+                        <Title headingLevel="h1" size="4xl">{props.location.state ? props.location.state.title : 'No matches'}</Title>
+                        <EmptyStateBody>
+                            {props.location.state ? props.location.state.description : 'No data to display'}
+                        </EmptyStateBody>
+                        <Button variant="primary" onClick={redirectHandler}>
+                            {props.location.state? props.location.state.buttonText : 'Navigate to home'}
```
Comment (cristianonicolai) [inline packages/management-console/src/components/Molecules/NoDataComponent/NoDataComponent.tsx]:
```suggestion
                            {props.location.state? props.location.state.buttonText : 'Go to Process Instances'}
```

File: packages/management-console/src/components/Molecules/ErrorComponent/ErrorComponent.tsx [hunk 5]
```diff
@@ -1,52 +1,56 @@
...
+            <EmptyStateIcon
+              icon={ExclamationCircleIcon}
+              size="md"
+              color="var(--pf-global--danger-color--100)" />
+            <Title headingLevel="h1" size="4xl">404 Error: page not found</Title>
+            <EmptyStateBody>
+              This page could not be found.
+            </EmptyStateBody>
+            <Button variant="primary" onClick={redirectHandler}>
+              Go to process instances
```
Comment (cristianonicolai) [inline packages/management-console/src/components/Molecules/ErrorComponent/ErrorComponent.tsx:47]:
```suggestion
              Go to Process Instances
```
@Sara4994 please review capital letter on the wording for all texts.

**Review by AjayJagan — APPROVED** [review]
looks good :+1:

**Pull request merged**